### DISCUSSION TITLE
How to route request to two different backend based on request header

### DISCUSSION THREAD

**Discussion by raagulrakesh** [nginx/kubernetes-ingress#5503] [General]
Hi;

I have doubt in route using request header in ingress

I am have two service as service-1 and service-2 as two backend and If I get header as service:1 it should redirect to service-1 backend as well as for service-2

**Comment by jasonwilliams14** [comment]
@raagulrakesh Take a look at `match` and `conditions`:

https://docs.nginx.com/nginx-ingress-controller/configuration/virtualserver-and-virtualserverroute-resources/#match

This allows you to route based on a header to a specific service.

**Reply by raagulrakesh** [reply]
Hi 
I am using only Ingress resource. Is there is any possibility to achieve this in Ingress resource

**Reply by jasonwilliams14** [reply]
You can do it with the ingress resource, but you would need to use snippets, allowing you to insert nginx configs directly into Ingress controller.
Then you would add the snippets to your Ingress resource to check for the incoming header, then route to the appropriate upstream (service) in kubernetes.
The CRD approach makes this easy to do and the above link has the capability built directly in to our project.

**Comment by tinhtq** [comment]
which ingress do you use

**Reply by raagulrakesh** [reply]
I am using this ingress-controller https://github.com/kubernetes/ingress-nginx

**Reply by vepatel** [reply]
Hi @raagulrakesh, this is different project so I'd recommend opening a discussion/issue there or look at https://docs.nginx.com/nginx-ingress-controller/installation/ingress-nginx/ if you're open to moving to this project

thanks!
